### ISSUE TITLE
Potentially wrong scheduler in train_text_to_image_sdxl.py?

### ISSUE THREAD

**Issue by christopher-beckham** [huggingface/diffusers#8281]
### Describe the bug

I see that in [train_text_to_image_sdxl.py](https://github.com/huggingface/diffusers/blob/main/examples/text_to_image/train_text_to_image_sdxl.py) the scheduler being used is `DDPMScheduler`, which -- as the name suggests -- is the specific diffusion formulation in [this paper](https://arxiv.org/abs/2006.11239). However, this contradicts the config of the actual SDXL model, it's using `EulerDiscreteScheduler` as you can see here: https://huggingface.co/stabilityai/stable-diffusion-xl-base-1.0/blob/main/model_index.json

`EulerDiscreteScheduler` (and other compatible schedulers) can't just be arbitrarily swapped in with `DDPMScheduler` because the former computes a noised sample via `x_t = x_0 + sigma_t*eps` (i.e. it implements variance exploding SDE) whilst DDPM is the following _variance preserving SDE_, and `x_t` is computed as follows:
<img width="368" alt="image" src="https://github.com/huggingface/diffusers/assets/2417792/14d02f74-5a16-415b-a2e1-63167262675c">

If `DDPMScheduler` was simply instantiated to just pull out the betas and derive sigmas from them that's fine (to compute the noisy sample from the exploding SDE), but if we look [here](https://github.com/huggingface/diffusers/blob/main/examples/text_to_image/train_text_to_image_sdxl.py#L1087) we can see that `scheduler.add_noise` is being used which means we're computing `x_t` based on the DDPM formulation.

In fact, from other SDXL-based scripts (e.g. [this](https://github.com/huggingface/diffusers/blob/main/examples/t2i_adapter/train_t2i_adapter_sdxl.py#L1207-L1224) and [this](https://github.com/huggingface/diffusers/blob/0ab63ff6478b7cc6b5ae0d46c7c386d476cfa87f/examples/dreambooth/train_dreambooth_lora_sdxl.py#L1673)) the variance exploding formulation is used, but with one extra caveat that the noisy inputs also get scaled with `noisy_model_input / ((sigmas**2 + 1) ** 0.5)` before being passed into the UNet (and this is _extremely_ important).

Therefore, my question is: is there a particular reason why `DDPMScheduler` is used here or is it just an oversight/bug? I have seen issues raised about this script in the past, including this one https://github.com/huggingface/diffusers/issues/4827

Thanks.

### Reproduction

n/a (asking a fundamental question related to the code)

### System Info

n/a

**Comment by bghira** [comment]
also, the Euler scheduler is used on the internal controlnet training script used by Hugging Face's Suraj Patil (cc @patil-suraj) and the word was that the initial samples looked better. but the long term results of training over several hundred thousand iterations is no better or different than using DDPM.

**Comment by christopher-beckham** [comment]
Thanks @bghira, indeed it was an oversight on my part that the scheduler specified in the config is intended to be the inference scheduler (not necessarily the training one). I understand that both can be different, but as far as the training scheduler is concerned it is counterintuitive to me if a model trained with DDPMScheduler (i.e. does mean scaling) is subsequently fine-tuned with a scheduler that doesn't (if you're implying that's the case with ControlNet script).

Otherwise, if SDXL was trained with `DDPMScheduler` then I have no issues fine-tuning that model on the same scheduler.

**Comment by bghira** [comment]
i'm also wanting more information on the matter, so i'm glad you started the discussion.

for perhaps a little more insight, it might be worth it to note that the DreamShaper series by Lykon has a recent version that requires DPM++ 2M Karras (i think the SDE variant) because he tuned it on that noise schedule.

it's entirely possible that training on DDPM makes the model more broadly compatible with other schedules at inference time.

**Comment by leeruibin** [comment]
I have the same question when I try to fineturning CosXL model. As they stated in [CosXL](https://huggingface.co/stabilityai/cosxl), CosXL employ Cosine-Continuous EDM VPred scheduler to turn the model. I follow the instruction of [train_dreambooth_lora_sdxl](https://github.com/huggingface/diffusers/blob/main/examples/dreambooth/train_dreambooth_lora_sdxl.py) to fineturn CosXL model as there are official implementation of edm_style_training by diffuser. However, the model perfermance will become terrible after I try to fineturn it.

I have tried EDMEulerScheduler and DDPMScheduler but none of them seems work well. I have on idea whether the EDMEulerScheduler is exactly the Cosine-Continuous EDM VPred scheduler mentioned by StablilityAI or not. They only release the model parameters and give a very simple statement. 

I would be grateful if you could give me some suggestions to make the fineturninig progress work normally. @bghira

**Comment by bghira** [comment]
see [this](https://huggingface.co/spaces/multimodalart/cosxl/blob/main/app.py) for cosxl demo code

**Comment by leeruibin** [comment]
Thanks for your reply. Actually, I have referred to this code in my training code.

In my setting, I can edit image successfully when I use instruction like "change the cup to red" before fineturning. However, when I try to fineturn the CosXL_edit model with [fusing/instructpix2pix-1000-samples](https://huggingface.co/datasets/fusing/instructpix2pix-1000-samples) dataset as a toy experiment, the model becomes weird. The loss can decrease normally but the editing fails and the output image contains many artifacts.

The training code is in [here](https://github.com/leeruibin/Attempt_CosXL_fineturning/blob/main/train_CosXLEdit_customized_data.py), and here are my fineturning results. I hope you could give me some advice to fix this problem. 

Thanks a lot!
![demo](https://github.com/huggingface/diffusers/assets/26408630/92411946-c491-4c23-96b8-8ff51eea3912)

**Comment by bghira** [comment]
wow, that is a totally new style of failure mode to me and i'm not sure what would cause that other than scheduler configuration details. i was thinking maybe you're experiencing an issue with unconditional space. are you using negative prompts at inference time, and, does reducing/increasing the step count change anything?

**Comment by bghira** [comment]
also, 50k steps on 1k images seems like quite a lot of training. have you tested earlier checkpoints?

**Comment by leeruibin** [comment]
I also believe this is caused by the wrong setting of the training scheduler. Currently, I follow the setting of [train_dreambooth_lora_sdxl](https://github.com/huggingface/diffusers/blob/main/examples/dreambooth/train_dreambooth_lora_sdxl.py) to set the training scheduler to EDMEulerScheduler, I am not sure whether this is the Cosine-Continuous EDM VPred schedule mentioned by stabilityAI. 

The inference setting of the fineturned model is the same as the first line. I use inference step 20, CFG=7, no negative prompt.

THe earlier checkpoints also fail. This is the evaluation result after 3K steps. The training batch size is 12.
![validation_3000_2598de14c4eff3b4df83](https://github.com/huggingface/diffusers/assets/26408630/ddf15f54-9364-4db5-9edc-af47f34731bb)

**Comment by christopher-beckham** [comment]
I'm a bit out of my depth here (I don't know anything about CosXL) but is there a citation or some reference for this "cosine continuous" schedule? Also, as the name suggests, are you finetuning with vpred instead of epsilon prediction?

**Comment by leeruibin** [comment]
Yes, I finetune it with vpred.

This is also the problem that bothers me. StabilityAI just released the model and said the CosXL model is a finetuned version of SDXL with Cosine-Continuous EDM VPred scheduler. Currently, they haven't given some details about this model as well as citations. 

I have tried this model and find this may be the most powerful model for conducting instruction-based editing to my knowledge. There is too little information in the official repo. Here is their model card [CosXL](https://huggingface.co/stabilityai/cosxl)

**Comment by christopher-beckham** [comment]
BTW we could maybe move this into its own separate discussion or issues since it's pertaining to CosXL.

I was originally planning on closing this issue since I had my own misunderstanding with what the appropriate scheduler is for SDXL. Now I know that it was trained as a DDPM model but I still find it odd that the inference scheduler is `EulerDiscreteScheduler` because that is actually based on Algorithm 2 of [EDM](https://arxiv.org/abs/2206.00364) (without the 2nd order step), but Algorithm 2 is actually assuming `s(t)=1` and `sigma(t) = t` (see Table 1 in their paper), therefore it's not actually the DDPM formulation which has its own specific `s(t)` and `sigma(t)`:

<img width="296" alt="image" src="https://github.com/huggingface/diffusers/assets/2417792/e1f2e718-1ee5-494f-b8eb-1423d1844722"><br />

Of course, things seem to work out empirically and the samples are still good, but I want to mention it anyway.
